test/NuGet.Core.Tests/NuGet.Configuration.Test/PackageNamespacesProviderTests.cs

```diff
@@ -0,0 +1,423 @@
+            packageSourceNamespaces.Should().HaveCount(1);
+            var packageSourceNamespace = packageSourceNamespaces.First();
+            packageSourceNamespace.Key.Should().Be("nuget.org");
+            packageSourceNamespace.Namespaces.Should().HaveCount(1);
+            packageSourceNamespace.Namespaces.First().Id.Should().Be("stuff");
+        }
+
+        [Fact]
+        public void GetPackageSourceNamespaces_WithMultipleConfigs_CombinesDifferentKeys()
+        {
+            // Arrange
+            using var mockBaseDirectory = TestDirectory.Create();
+            var configPath1 = Path.Combine(mockBaseDirectory, "NuGet.Config");
+            SettingsTestUtils.CreateConfigurationFile(configPath1, @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageNamespaces>
+        <packageSource key=""nuget.org"">
+            <namespace id=""stuff"" />
+        </packageSource>
+    </packageNamespaces>
+</configuration>");
+            var configPath2 = Path.Combine(mockBaseDirectory, "NuGet.Config.2");
+            SettingsTestUtils.CreateConfigurationFile(configPath2, @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageNamespaces>
+        <packageSource key=""contoso"">
+            <namespace id=""stuff2"" />
+        </packageSource>
+    </packageNamespaces>
+</configuration>");
+            var settings = Settings.LoadSettingsGivenConfigPaths(new string[] { configPath1, configPath2 });
+
+            // Act & Assert
+            var namespaceProvider = new PackageNamespacesProvider(settings);
+            var packageSourceNamespaces = namespaceProvider.GetPackageSourceNamespaces();
+            packageSourceNamespaces.Should().HaveCount(2);
+
+            var contosoNamespace = packageSourceNamespaces.First();
+            contosoNamespace.Key.Should().Be("contoso");
+            contosoNamespace.Namespaces.Should().HaveCount(1);
+            contosoNamespace.Namespaces.First().Id.Should().Be("stuff2");
+
+            var nugetOrgNamespace = packageSourceNamespaces.Last();
+            nugetOrgNamespace.Key.Should().Be("nuget.org");
+            nugetOrgNamespace.Namespaces.Should().HaveCount(1);
+            nugetOrgNamespace.Namespaces.First().Id.Should().Be("stuff");
+        }
+
+        [Fact]
+        public void GetPackageSourceNamespaces_WithMultipleConfigs_RespectsClearTag()
+        {
+            // Arrange
+            using var mockBaseDirectory = TestDirectory.Create();
+            var configPath1 = Path.Combine(mockBaseDirectory, "NuGet.Config");
+            SettingsTestUtils.CreateConfigurationFile(configPath1, @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageNamespaces>
+        <clear />
+        <packageSource key=""nuget.org"">
+            <namespace id=""stuff"" />
+        </packageSource>
+    </packageNamespaces>
+</configuration>");
+            var configPath2 = Path.Combine(mockBaseDirectory, "NuGet.Config.2");
```

Comment:
Having 2 nuget.config files are same directory may not resemble actual world usage?
Can you put them in separate directories? Now hard to see which is actual effective nuget.config in working directory.

Reply:
I don't think that's necessary. 
We're not testing the config priority based on location here, which is why I'm using the method that takes priority based on argument order.